File: java/json/src/main/java/com/microsoft/azure/schemaregistry/kafka/json/JsonSerializerException.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.azure.schemaregistry.kafka.json;

import org.apache.kafka.common.errors.SerializationException;

/**
 * Custom error class for exceptions thrown in Json serialization/deserialization steps
 */
public class JsonSerializerException extends SerializationException {
    /**
     * Constructor with message only
     * @param errorMessage Brief explination of exception source.
     */
    public JsonSerializerException(String errorMessage) {
        super(errorMessage);
    }

    /**
     * Constructor with message and throwable error
     * @param errorMessage Brief explination of exception source.
     * @param err Throwable error object
     */
    public JsonSerializerException(String errorMessage, Throwable err) {
        super(errorMessage, err);
    }
}

File: csharp/json/src/JsonSerializerException.cs
//------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//------------------------------------------------------------

namespace Microsoft.Azure.Kafka.SchemaRegistry.Json
{
    using System;

    public class JsonSerializerException : Exception
    {
        public JsonSerializerException()
        {
        }

        public JsonSerializerException(string message) : base(message)
        {
        }

        public JsonSerializerException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}
